setup.py

```diff
@@ -56,10 +59,40 @@ def run_tests(self):
         sys.exit(errno)
 
 
+# delay import of numpy until it is installed
```

Comment:
It's not obvious to me what is going on here. Could you please leave a longer comment here or in the docs?
Numpy is now a required dependency?
And this delayed import is needed because `setup.py` is executed by `pip` before `numpy` as a dependency is installed?

Reply:
Yes, Numpy is now a required dependency. This one of things we discussed and you preferred that over the more complex code I had before, which kept numpy optional. Do you remember?

Reply:
It is  not clear to me whether this code here is strictly necessary. The general idea is to delay the import of numpy until it is installed by pip as part of the requirements if you don't already have it. I have no clear understanding how setup.py interplays with pip. I just copied this more or less. There is not much more to comment here.

Reply:
I remember the discussion, but to be honest not the important detail whether compile-time linking against Numpy is needed / useful or not.

Is compile-time linking required to support the second cost function type you are adding here? I think this is the case, even if internally in iminuit everything still goes through Python types, i.e. there is no performance benefit to introduce this.

Ideally there would be a few lines at
http://iminuit.readthedocs.io/en/latest/installation.html
describing this, and then maybe a link to that description from the `setup.py` file.
Or you decide which docs are better suited as code comments or for the docs, so that the design and build is clear to users and devs.

Reply:
> I remember the discussion, but to be honest not the important detail whether compile-time > linking against Numpy is needed / useful or not.
> Is compile-time linking required to support the second cost function type you are adding
> here? I think this is the case, even if internally in iminuit everything still goes through
> Python types, i.e. there is no performance benefit to introduce this.

Yes, it must be exactly like this. We discussed all of that and I will not discuss it again here, it is tiring to repeat the same technical arguments over and over.

I will remove the lazy numpy import if that makes you happy.

Reply:
I'm just making small suggestions to add a few lines documenting how the new solutions work. Up to you to take them or just merge.